pkg/scheduler/framework/parallelize/parallelism.go

```diff
@@ -51,6 +52,12 @@ func chunkSizeFor(n, parallelism int) int {
 }
 
 // Until is a wrapper around workqueue.ParallelizeUntil to use in scheduling algorithms.
-func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc) {
-	workqueue.ParallelizeUntil(ctx, p.parallelism, pieces, doWorkPiece, workqueue.WithChunkSize(chunkSizeFor(pieces, p.parallelism)))
+func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc, operations ...string) {
+	withMetrics := func(piece int) {
+		metrics.Goroutines.WithLabelValues(operations...).Inc()
```

Comment:
this looks dangerous.
What happens if you pass a number of operations different than 1? Does that cause a panic?

Reply:
+1 to reverting and comment

Reply:
another option is to have a second function `Parallelizer.UntilWithOperation`

Reply:
Or pull the operation into the context? 🤷

Reply:
👌 Reverted and add the comment about `operation`.